test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ButtonTagHelperTest.cs

```diff
@@ -0,0 +1,381 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.AspNetCore.Routing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    public class ButtonTagHelperTest
+    {
+        [Fact]
+        public async Task ProcessAsync_GeneratesExpectedOutput()
+        {
+            // Arrange
+            var expectedTagName = "not-button";
+            var metadataProvider = new TestModelMetadataProvider();
+
+            var tagHelperContext = new TagHelperContext(
+                allAttributes: new TagHelperAttributeList
+                {
+                    { "id", "mybutton" },
+                    { "asp-route-name", "value" },
+                    { "asp-action", "index" },
+                    { "asp-controller", "home" },
+                },
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+            var output = new TagHelperOutput(
+                expectedTagName,
+                attributes: new TagHelperAttributeList
+                {
+                    { "id", "mybutton" },
+                },
+                getChildContentAsync: (useCachedResult, encoder) =>
+                {
+                    var tagHelperContent = new DefaultTagHelperContent();
+                    tagHelperContent.SetContent("Something Else");
+                    return Task.FromResult<TagHelperContent>(tagHelperContent);
+                });
+            output.Content.SetContent("Something");
+
+            var urlHelper = new Mock<IUrlHelper>();
+            urlHelper
+                .Setup(mock => mock.Action(It.IsAny<UrlActionContext>())).Returns("home/index").Verifiable();
+
+            var urlHelperFactory = new Mock<IUrlHelperFactory>();
+            urlHelperFactory
+                .Setup(f => f.GetUrlHelper(It.IsAny<ActionContext>()))
+                .Returns(urlHelper.Object);
+
+            var htmlGenerator = new TestableHtmlGenerator(metadataProvider, urlHelper.Object);
+            var viewContext = TestableHtmlGenerator.GetViewContext(model: null,
+                                                                   htmlGenerator: htmlGenerator,
+                                                                   metadataProvider: metadataProvider);
```

Comment:
Put the first argument on its own line too and don't get too fancy w/ indentation:
``` c#
var viewContext = TestableHtmlGenerator.GetViewContext(
    model: null,
    htmlGenerator: htmlGenerator,
    metadataProvider: metadataProvider);
```